frontend/main-app/src/components/FormViewer/httpUtils.tsx

```diff
@@ -0,0 +1,53 @@
+import { getApiUrl } from "../Login/LoginUtils";
+import axios from "axios";
+
+export const getFormSchema = async (formId: string, setFormSchema) => {
+  const apiUrl = getApiUrl();
+  const targetUrl = `${apiUrl}/api/forms/${formId}`;
+
+  try {
+    const response = await axios.get(targetUrl);
+    setFormSchema(response.data);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const formatSubmission = (formSchema, formData) => {
+  const responses = [];
+  let i = 0;
+
+  // Create a response object for every question in the submitted form
+  for (const key in formData) {
+    const response = {
+      name: formSchema.components[i].name,
+      order: formSchema.components[i].order,
+      label: formSchema.components[i].label,
+      response: formData[key],
+    };
+
+    responses.push(response);
+    i++;
+  }
+
+  return {
+    formId: formSchema.id,
+    submitted: new Date(),
+    responses: responses,
+  };
+};
```

Comment:
Your solution is good, but I think utilizing the mapping's potential here it makes it even more readable.
Haven't tested that this works, so maybe I missing something here.
```suggestion
const formatSubmission = (formSchema, formData) => {
  const responses = formSchema.components.map((component, index) => ({
    name: component.name,
    order: component.order,
    label: component.label,
    response: formData[index],
  }));

  return {
    formId: formSchema.id,
    submitted: new Date(),
    responses,
  };
};
```

Reply:
That's really smooth, much more readable! 💯 
This is how it turned out after some testing:

````
const formatSubmission = (formSchema, formData) => {
  const responses = formSchema.components.map((component) => ({
    name: component.name,
    order: component.order,
    label: component.label,
    response: formData[component.name] || "No response",
  }));

  return {
    formId: formSchema.id,
    submitted: new Date(),
    responses,
  };
};
````